lazy-imports.html

```diff
@@ -0,0 +1,64 @@
+<!--
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="../polymer/polymer.html">
+
+<!--
+`lazy-imports`
+Declarative lazy imports
+
+@demo demo/index.html
+-->
+<script>
+
+/** @polymerBehavior */
+var LazyImportsBehavior = {
+  __loadedImports: {},
+  importLazyGroup: function(group) {
+    var domModule = Polymer.DomModule.import(this.localName);
+    var groupAttribute = '';
+    if (group) {
+      groupAttribute = `[group=${group}]`;
+    }
+    var query = `link${groupAttribute}[rel='lazy-import'][href]`
+    var imports = domModule.querySelectorAll(query);
+    var promises = [];
+    for (var i = 0; i < imports.length; i++) {
+      promises.push(new Promise((resolve, reject) => {
+        var href = imports[i].getAttribute('href');
+        if (!href || href.length == 0) {
+          return;
+        }
+        var onLoad = function() {
+          resolve({loaded: href});
+        }
+        var onFailure = function() {
+          resolve({failed: href});
+        }
+        var resolved = this.resolveUrl(href);
+        if (!this.__loadedImports[resolved]) {
+          this.__loadedImports[resolved] = true;
+          this.importHref(this.resolveUrl(href), onLoad, onFailure);
```

Comment:
You can reuse `resolved` instead of computing again with `this.resolveUrl(href)`